### ISSUE TITLE
Ghidra fails to decompile function: 'Low-level Error: Unable to merge address forced indirect'

### ISSUE THREAD

**Issue by uniboi** [NationalSecurityAgency/ghidra#5588]
**Describe the bug**
Decompiling the function `FUN_180059a30` throws the error `Low-level Error: Unable to merge address forced indirect`.
The function is part of a custom type checking compiler for a modified version of [squirrel](https://www.squirrel-lang.org/). Because the function is so big I wasn't able to narrow the issue down.
IDA is able to generate pseudocode for the function.

**To Reproduce**
1. Download the binary ([server.dll.zip](https://github.com/NationalSecurityAgency/ghidra/files/12138741/server.dll.zip))
2. Decompile the function `FUN_180059a30` at `0x180059a30`
3. After a few seconds the decompiler should throw the error `Low-level Error: Unable to merge address forced indirect`

**Expected behavior**
The decompiler is able to generate pseudocode

**Attachments**
[server.dll.zip](https://github.com/NationalSecurityAgency/ghidra/files/12138741/server.dll.zip)

**Environment:**
 - OS: Windows 10
 - Java Version: 17.0.6
 - Ghidra Version: 10.3
 - Ghidra Origin: official GitHub distro

**Comment by LukeSerne** [comment]
The error is thrown by `Merge::mergeIndirect` in `merge.cc:881` (in the source of Ghidra 10.3, the file has seen some changes since, so line numbers have changed compared to current `master`). I added `indop->printDebug(std::cout);` just before the error is thrown, compiled `decomp_dbg`, created an `.xml` of the function (which I zipped and attached to this comment as well) and loaded that in `decomp_dbg`. Then I tried decompiling the function again and reproduced the problem. Additionally, the offending `INDIRECT` operation was printed: `0x00018005b169:d742: s0x00000010(0x00018005b169:d742) = u0x10004f15(0x00018005b169:112cd) [] i0x00018005b169:2aab(free)` -> i.e. by one of the PCode operations that is produced by the instruction at address `0x18005b169`.

Going back to Ghidra, I noticed that this address is undefined data. This is because when I initially disassembled at address `0x180059a30`, Ghidra was unable to resolve the switch table that is at address `0x180059a8e`. When manually going through each of the cases of the switch table, and disassembling there, I noticed that all of the branches are decompiled correctly. Even the `CALL` instruction at `0x18005b169` decompiles correctly when decompiled as part of the (fake) function starting at `0x180059a91`.

From reading the relevant code in `merge.cc` (see below), it seems that the problem is that `indop->getIn(0)->getHigh()` and `outvn->getHigh()` overlap at line 879. I am not sure why that happens or what causes this.

https://github.com/NationalSecurityAgency/ghidra/blob/104d9592485c2fb6c7ab5177c71d8d70140a3d03/Ghidra/Features/Decompiler/src/decompile/cpp/merge.cc#L845-L882

Although I could not track down the root cause of this problem, I hope my initial analysis will be useful in the further investigation of this issue.

[server.xml.zip](https://github.com/NationalSecurityAgency/ghidra/files/12139961/server.xml.zip)
